pkg/sdk/validation/typeinstance.go

```diff
@@ -0,0 +1,143 @@
+}
+
+// ValidateTypeInstanceToUpdate  is responsible for validating TypeInstance which exists and will be updated.
+func ValidateTypeInstanceToUpdate(ctx context.Context, client TypeInstanceValidationHubClient, typeInstanceToUpdate []graphqllocal.UpdateTypeInstancesInput) (Result, error) {
+	var typeInstanceIds []string
+	idToTypeNameMap := map[string]string{}
+	for _, ti := range typeInstanceToUpdate {
+		typeInstanceIds = append(typeInstanceIds, ti.ID)
+	}
+
+	typeInstancesTypeRef, err := client.FindTypeInstancesTypeRef(ctx, typeInstanceIds)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while finding TypeInstance Type reference")
+	}
+
+	typeRefCollection := TypeRefCollection{}
+	for id, typeReference := range typeInstancesTypeRef {
+		name := getManifestPathWithRevision(typeReference.Path, typeReference.Revision)
+		typeRefCollection[name] = TypeRef{
+			TypeRef: types.TypeRef{
+				Path:     typeReference.Path,
+				Revision: typeReference.Revision,
+			},
+		}
+		idToTypeNameMap[id] = name
+	}
+
+	var typeInstanceCollection []*typeInstanceData
+	for _, ti := range typeInstanceToUpdate {
+		if ti.TypeInstance == nil {
+			continue
+		}
+		typeInstanceCollection = append(typeInstanceCollection, &typeInstanceData{
+			id:                  ti.ID,
+			value:               ti.TypeInstance.Value,
+			typeRefWithRevision: idToTypeNameMap[ti.ID],
+		})
+	}
+
+	schemasCollection, err := ResolveTypeRefsToJSONSchemas(ctx, client, typeRefCollection)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while resolving TypeRefs to JSON Schemas")
+	}
+
+	return validateTypeInstances(schemasCollection, typeInstanceCollection)
+}
+
+func validateTypeInstances(schemaCollection SchemaCollection, typeInstanceCollection []*typeInstanceData) (Result, error) {
+	resultBldr := NewResultBuilder("Validation TypeInstances")
+
+	for _, ti := range typeInstanceCollection {
+		if _, ok := ti.value.(map[string]interface{}); !ok {
+			return Result{}, errors.New("could not create map from TypeInstance Value")
+		}
+		valuesJSON, err := json.Marshal(ti.value)
+		if err != nil {
+			return Result{}, errors.Wrap(err, "while converting TypeInstance value to JSON bytes")
+		}
+		if _, ok := schemaCollection[ti.typeRefWithRevision]; !ok {
+			return Result{}, fmt.Errorf("could not find Schema for type %s", ti.typeRefWithRevision)
+		}
+
+		schemaLoader := gojsonschema.NewStringLoader(schemaCollection[ti.typeRefWithRevision].Value)
+		dataLoader := gojsonschema.NewBytesLoader(valuesJSON)
+
+		result, err := gojsonschema.Validate(schemaLoader, dataLoader)
+		if err != nil {
+			return nil, errors.Wrap(err, "while validating JSON schema for TypeInstance")
+		}
+		if !result.Valid() {
+			for _, err := range result.Errors() {
+				msg := ""
+				if ti.alias != nil {
+					msg = fmt.Sprintf("TypeInstance with alias %s", *ti.alias)
+				} else if ti.id != "" {
+					msg = fmt.Sprintf("TypeInstance with id %s", ti.id)
+				}
```

Comment:
Theoretically, in this scope of the validation function, there could be a case that a given TypeInstance contains both Alias and ID, right? So please use StringBuilder to build proper message taking into account that 🙂

Reply:
Right, theoretically it could have both. I added a separate method and now the output is as:  
```
Validation TypeInstances "TypeInstance(ID: some-id,Alias: aws-creds)":
    * (root): name is required
    * (root): Additional property host is not allowed
```